Update MerlinAU.sh Unmount

ExtremeFiretop opened the pull request:
Again same general improvement as last round.
This command just is simpler than the function.

1. The function displays no text to me and has not once, while trying to unmount the drive. 
(I would expect to see the printf "\nUnmounting USB-attached drives. Please wait...\n")
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/56fbe816-5431-40ed-8f7c-8bddd1638a69)


2. I am not a fan of how your function attempts to unmount my UBIFS (which stores my JFFs).
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/f736f504-c54b-4aee-bec8-1218931c9128)

3. Finally it never worked as pictured below:
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/56fbe816-5431-40ed-8f7c-8bddd1638a69)

![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/ec639499-e65f-462e-9164-e194bc538e24)


----Setup and tests-----
Cleanly formatted USB:
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/483603cf-7786-4ec4-b7b2-278aa7d412ac)

Move the stop flash right after the unmount function.. To be able to test and catch the text or check the USB afterwards:
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/fca5bd34-2738-4369-830d-2c5418cdb539)

No text is displayed and no USB is unmounted and this again is cleanly formatted not being used actively for any real purpose
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/7c840f92-9382-45be-b40c-5015b9cb84df)


In comparison the command provided instantly unmounted samba and the USB:
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/55369efc-cebf-41e3-8df1-e7b65ced8629)
![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/d2e45f44-58ef-423c-bf0a-bad24e9a470d)

ExtremeFiretop commented:
Tested and functional.

Did 2 successful upgrades from 388.5.beta1 to production 388.6 without a hitch or errors with option 3 using this method.

ExtremeFiretop commented:
Now 4 successful back to back upgrades using this method from beta1 to production without a hitch or issue with option 3 post-reboot.

Which means my issue was likely related to the LED's or this.
Either way, even if it wasn't this, this never functioned for me as described in detail above, so lets go with something with that works with UBIFS routers and the older JFFs routers

ExtremeFiretop commented:
Discussions around the use of: "ejusb"

https://www.snbforums.com/threads/usb-dcl-disk-check-log-constantly-recovering-journal.73336/post-697992

https://www.snbforums.com/threads/usb-dcl-disk-check-log-constantly-recovering-journal.73336/post-697793

https://www.snbforums.com/threads/solved-ac86u-possible-memory-leaks-aiprotection-dcd-crashing-is-guilty.55416/

https://www.snbforums.com/threads/solved-ac86u-possible-memory-leaks-aiprotection-dcd-crashing-is-guilty.55416/post-471194

https://www.snbforums.com/threads/tutorial-native-transmission-on-asuswrt-routers-through-optware.8696/post-142761

Works on AC86U as confirmed by the links above in the forums (that was my only concern).
Merged.

ExtremeFiretop commented:
Fixed the path to ejusb in commit:

https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/111/commits/d5fe282528698e90c89863dd934ad1d26df1f4ec

Martinski4GitHub commented:
> 
>  2. I am not a fan of how your function attempts to unmount my UBIFS (which stores my JFFs).
> 
> ...
> 
> ... your function ... does not function correctly for me and attempted to unmount my UBIFS (and possibly even succeeded the first time I had issues)
>
If I were a really paranoid guy, I'd say that you were making this stuff up & leave it at that; but based on all our previous interactions & exchanges for the past 3 months, I believe instead, in good faith, that something unusual was going on with your router, or the method you were using to test is seriously flawed because I know with 100% certainty that the "Unmount" function I wrote does ***NOT*** attempt to do what you claimed above.

I tested that function in DEBUG mode on 3 different routers (the actual "unmount" call was commented out & replaced with a DEBUG print statement showing every mount point selected), so I know perfectly well that the function tries to unmount ***ONLY*** the existing USB drives, and nothing else. Now, worst-case scenario, the function may not find any USB drives, as in your specific router (I suspect because it's not **EXT**n or **NTFS** formatted), but it doesn't go beyond USB drive mount points.

I don't know what you did to make it appear as though the function is attempting to do what you claimed in your messages, but I certainly know that's not correct. Guaranteed. I even said in one of my messages that I was testing it thoroughly because "we don't want to accidentally unmount JFFS!!" (see comment here: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/96#issuecomment-1913781449).

If you don't trust my word, don't take it. You can check it yourself by copying the 2 lines that select the mounted devices from the original function (see here : https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/110/files) and running them on the same router, under the exact same conditions (same USB drive) like this (including the enclosing curly braces):

`
{`
`    mountPointRegExp="^/dev/sd[a-z][0-9]+.* /tmp/mnt/.*"`
`    mountedDevices="$(grep -E "$mountPointRegExp" /proc/mounts | awk -F ' ' '{print $1}')"`
`    echo "MOUNTED DEVICES:"`
`    echo "[$mountedDevices]"`
`}
`

Then post the results of the above code here.
As an example, here's a screenshot from my router:

![MountedDevices](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/119833648/7a175d09-7fda-4346-bae2-8fb110f097a3)

ExtremeFiretop commented:
> > 2. I am not a fan of how your function attempts to unmount my UBIFS (which stores my JFFs).
> > 
> > ...
> > ... your function ... does not function correctly for me and attempted to unmount my UBIFS (and possibly even succeeded the first time I had issues)
> 
> If I were a really paranoid guy, I'd say that you were making this stuff up & leave it at that; but based on all our previous interactions & exchanges for the past 3 months, I believe instead, in good faith, that something unusual was going on with your router, or the method you were using to test is seriously flawed because I know with 100% certainty that the "Unmount" function I wrote does _**NOT**_ attempt to do what you claimed above.
> 
> I tested that function in DEBUG mode on 3 different routers (the actual "unmount" call was commented out & replaced with a DEBUG print statement showing every mount point selected), so I know perfectly well that the function tries to unmount _**ONLY**_ the existing USB drives, and nothing else. Now, worst-case scenario, the function may not find any USB drives, as in your specific router (I suspect because it's not **EXT**n or **NTFS** formatted), but it doesn't go beyond USB drive mount points.
> 
> I don't know what you did to make it appear as though the function is attempting to do what you claimed in your messages, but I certainly know that's not correct. Guaranteed. I even said in one of my messages that I was testing it thoroughly because "we don't want to accidentally unmount JFFS!!" (see comment here: [#96 (comment)](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/96#issuecomment-1913781449)).
> 
> If you don't trust my word, don't take it. You can check it yourself by copying the 2 lines that select the mounted devices from the original function (see here : https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/110/files) and running them on the same router, under the exact same conditions (same USB drive) like this (including the enclosing curly braces):
> 
> ` {` ` mountPointRegExp="^/dev/sd[a-z][0-9]+.* /tmp/mnt/.*"` ` mountedDevices="$(grep -E "$mountPointRegExp" /proc/mounts | awk -F ' ' '{print $1}')"` ` echo "MOUNTED DEVICES:"` ` echo "[$mountedDevices]"` `} `
> 
> Then post the results of the above code here. As an example, here's a screenshot from my router:
> 
> ![MountedDevices](https://private-user-images.githubusercontent.com/119833648/301369187-7a175d09-7fda-4346-bae2-8fb110f097a3.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY3NTg3MDIsIm5iZiI6MTcwNjc1ODQwMiwicGF0aCI6Ii8xMTk4MzM2NDgvMzAxMzY5MTg3LTdhMTc1ZDA5LTdmZGEtNDM0Ni1iYWUyLThmYjExMGYwOTdhMy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwMVQwMzMzMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjI5MmNjNjhkNDAwMjJhNzc0MGQzMWQxYzY1MjE4ZDg5ZTU0OTMzNTliMTE1MDA1OTg0MDI5NjhlNjk0NWQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.WfoKoHDG-zlUVFSWAkmNWqwyyQML4YoJxtOihAH15BA)

Device functions perfectly well and is right now, and so has all my testing up to thus point and again today:
Today, I did test again at your request, and your partly right, now it only displays sda1 instead of all my mount points:

![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/276ab848-9d36-42d4-98e3-8aaf4252c6a4)

It's possible I typo'd something or missed something in my first test function.

However, it still yet, does not unmount my USB when I use your function:

Backing up to USB:

![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/ec0b0ba1-9670-4098-a075-f4f452c4f432)

"Unmount USB" (Nothing happened)

![image](https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/assets/1971404/632fa2f3-7b09-4e88-b23b-ae47b6f11da9)

Refreshed with Control+F5 all day long and it's still mounted, do ejusb -1 0 -u 1 and it's gone instantly.
ejusb is the function the router uses itself at shutdown, do you not trust a function the router uses every time it shuts down? If I were really paranoid I'd be asking why we weren't using this from the start!

I brought this up yesterday, but you wanted to focus on the sleep so, I took it upon myself to resolve the issue for myself. I was not going to run a production script which didn't function in my environment as I expected.

If i have the issue, others will too, and that's enough thinking for me to implement something stronger as I mentioned above and yesterday already. Now if you want to help troubleshoot why your function is not unmounting anything for me? Then feel free, but the same way as the $isInactive was broken and I fixed it quickly, I did the same here.

Sometimes making a judgement call is important, and the rights ones are important as well, and in this case, again whether or not the function attempts to unmount my UBIFS, (maybe that was my mistake), but it still does not work, so I made the judgement call to use something else.

Feel free to troubleshoot, send me more things to try, etc, but as of now, this works, and this gives us time to continue to troubleshoot, etc. without having something dangling over my head in production.

I really think we are focusing in on the wrong things, we should be focusing in on things we should be trying to narrow down the issue and resolve it, instead of narrowing down on the fact that you think I'm making stuff up. 

If this is the angle this will take again tonight I'm simply not interested in the drama, I've changed many things for you without a seconds thought, I've asked for one sleep to be changed and it turned into 42 pages, now I did a quick "emergency" change to resolve an issue on my own personal device, and I hope you can just simply move on to the next question which is what is wrong and how can we solve it. 

I've made it endlessly clear I cannot endorse a function which does not work for me. This is one of them. Lets work on fixing it, instead of the fact that I replaced a function and I've already provided evidence of that above and mentioned it across multiple PRs.

ExtremeFiretop commented:
And to be fair, I also give you the benefit of the doubt over working the last few months that you simply didn't know about this function and don't mean any harm.

But we also use the routers built in move functions like mv, I don't think we should be building our own functions to do what mv does either, so I think in this case ejusb is perfectly usable to keep things safe while we figure the rest out.

Martinski4GitHub commented:
> 
> Refreshed with Control+F5 all day long and it's still mounted, do ejusb -1 0 -u 1 and it's gone instantly. ejusb is the function the router uses itself at shutdown, do you not trust a function the router uses every time it shuts down? If I were really paranoid I'd be asking why we weren't using this from the start!
> 
> I brought this up yesterday, but you wanted to focus on the sleep so, I took it upon myself to resolve the issue for myself. I was not going to run a production script which didn't function in my environment as I expected.
> 
> If i have the issue, others will too, and that's enough thinking for me to implement something stronger as I mentioned above and yesterday already. Now if you want to help troubleshoot why your function is not unmounting anything for me? Then feel free, but the same way as the $isInactive was broken and I fixed it quickly, I did the same here.
> 
> Sometimes making a judgement call is important, and the rights ones are important as well, and in this case, again whether or not the function attempts to unmount my UBIFS, (maybe that was my mistake), but it still does not work, so I made the judgement call to use something else.
> 
> Feel free to troubleshoot, send me more things to try, etc, but as of now, this works, and this gives us time to continue to troubleshoot, etc. without having something dangling over my head in production.
> 
> I really think we are focusing in on the wrong things, we should be focusing in on things we should doing to try to narrow down the issue and resolve it, instead of narrowing down on the fact that you think I'm making stuff up.
> 
> If this is the angle this will take again tonight I'm simply not interested in the drama, I've changed many things for you without a seconds thought, I've asked for one sleep to be changed and it turned into 42 pages, now I did a quick "emergency" change to resolve an issue on my own personal device, and I hope you can just simply move on to the next question which is what is wrong and how can we solve it.
> 
> I've made it endlessly clear I cannot endorse a function which does not work for me. This is one of them. Lets work on fixing it, instead of the fact that I replaced a function and I've already provided evidence of that above and mentioned it across multiple PRs.
>
You're misunderstanding my response. I'm not contesting at all the decision to replace the function. It's fine & I'm good with that; AFAICT, the "**ejusb**" command seems to work well, and that's what matters in the end.

My only point was that you were claiming that the function I wrote was doing something detrimental to the system, based on your testing results, which were *not* correct. I've already agreed that it doesn't find your USB drives on your router, and that's certainly a problem. But again, that's different from saying it was attempting to unmount the JFFS/UBIFS partition, which I know is absolutely false. That's the one & only point I'm trying to clarify here. It's a technical detail that I felt needed to be made.

I get the feeling that our trust in each other has been seriously eroded since the incident from yesterday & now this. You no longer trust my work enough to even consider the possibility that perhaps you were doing something wrong (something as simple as a typo) and didn't bother to double-check or triple-check to make sure that your testing methodology was solid and your test results were sound WRT the JFFS/UBIFS partition.

I don't feel comfortable collaborating with someone who doesn't have sufficient trust in my coding skills to first check & validate his own testing methods & his own results before jumping to the conclusion that the code has a serious flaw. It's not that I don't make mistakes (I certainly do), and that's why I've asked you before to review and check my PRs. That's standard SOP in S/W Dev. However, we also need to trust each other enough to make sure that you don't jump the gun and conclude that some new code is not only "just not working" but it's also detrimentally affecting the system.

Yes, this is the "drama" I didn't want in the first place. Too bad it got to this point.

Good Luck.

ExtremeFiretop commented:
> > Refreshed with Control+F5 all day long and it's still mounted, do ejusb -1 0 -u 1 and it's gone instantly. ejusb is the function the router uses itself at shutdown, do you not trust a function the router uses every time it shuts down? If I were really paranoid I'd be asking why we weren't using this from the start!
> > I brought this up yesterday, but you wanted to focus on the sleep so, I took it upon myself to resolve the issue for myself. I was not going to run a production script which didn't function in my environment as I expected.
> > If i have the issue, others will too, and that's enough thinking for me to implement something stronger as I mentioned above and yesterday already. Now if you want to help troubleshoot why your function is not unmounting anything for me? Then feel free, but the same way as the $isInactive was broken and I fixed it quickly, I did the same here.
> > Sometimes making a judgement call is important, and the rights ones are important as well, and in this case, again whether or not the function attempts to unmount my UBIFS, (maybe that was my mistake), but it still does not work, so I made the judgement call to use something else.
> > Feel free to troubleshoot, send me more things to try, etc, but as of now, this works, and this gives us time to continue to troubleshoot, etc. without having something dangling over my head in production.
> > I really think we are focusing in on the wrong things, we should be focusing in on things we should doing to try to narrow down the issue and resolve it, instead of narrowing down on the fact that you think I'm making stuff up.
> > If this is the angle this will take again tonight I'm simply not interested in the drama, I've changed many things for you without a seconds thought, I've asked for one sleep to be changed and it turned into 42 pages, now I did a quick "emergency" change to resolve an issue on my own personal device, and I hope you can just simply move on to the next question which is what is wrong and how can we solve it.
> > I've made it endlessly clear I cannot endorse a function which does not work for me. This is one of them. Lets work on fixing it, instead of the fact that I replaced a function and I've already provided evidence of that above and mentioned it across multiple PRs.
> 
> You're misunderstanding my response. I'm not contesting at all the decision to replace the function. It's fine & I'm good with that; AFAICT, the "**ejusb**" command seems to work well, and that's what matters in the end.
> 
> My only point was that you were claiming that the function I wrote was doing something detrimental to the system, based on your testing results, which were _not_ correct. I've already agreed that it doesn't find your USB drives on your router, and that's certainly a problem. But again, that's different from saying it was attempting to unmount the JFFS/UBIFS partition, which I know is absolutely false. That's the one & only point I'm trying to clarify here. It's a technical detail that I felt needed to be made.
> 
> I get the feeling that our trust in each other has been seriously eroded since the incident from yesterday & now this. You no longer trust my work enough to even consider the possibility that perhaps you were doing something wrong (something as simple as a typo) and didn't bother to double-check or triple-check to make sure that your testing methodology was solid and your test results were sound WRT the JFFS/UBIFS partition.
> 
> I don't feel comfortable collaborating with someone who doesn't have sufficient trust in my coding skills to first check & validate his own testing methods & his own results before jumping to the conclusion that the code has a serious flaw. It's not that I don't make mistakes (I certainly do), and that's why I've asked you before to review and check my PRs. That's standard SOP in S/W Dev. However, we also need to trust each other enough to make sure that you don't jump the gun and conclude that some new code is not only "just not working" but it's also detrimentally affecting the system.
> 
> Yes, this is the "drama" I didn't want in the first place. Too bad it got to this point.
> 
> Good Luck.

No trust was lost, I still believe you to be a great coder and many years ahead of others around you in your field surely. I just don't need every change I make questioned like I'm wrong or making stuff up when I have valid reason for the change.

I am a big enough man to admit when I was wrong, and I've already done that above by saying your function is partly functioning and I may have mistested something yesterday.

However I don't need long responses and drama over the change, I need a problem solver, I need someone that goes, issue? Where? And how can we fix it? Instead of "well actually nah nah, it didn't do exactly that you said down to the dot and dash in your response."

I'm more than happy to chat when things are working well, such as now, but when they need to be fixed I need someone that can focus on the issue at hand and not take a slight mis-test or slight misworded response and blow it out of the water. If you wish to avoid drama I recommend you don't take every response as drama and take them as issues to solve. Because every message I've ever is exactly that.

The only response I needed here was "can you retest this?" Or "I understand it doesn't work for you but how can we solve it?" Anything more from your side is drama and fluff I don't need as well. It is sad it came to this point when I know I'm working with a really skilled older gentleman that is taking small changes and small messages to heart. Not once did I make a personal attack, not once did I say anything was done or said wrong intentionally.

I work in the government with people the age of 65 retiring all around me, and I'm not used to someone that takes such stuff to heart, they look for problems to solve and where corrections can be made. Facts.

I can also make mistakes and have, that's why when you tell me you changed something or want me to change something, I just do it. I trust. I don't turn it into 42 comments long. If anything, it's clear the lost in trust was with your side yesterday.

Today you could of looked for a correction by having me retest, and then ask for extra info on the testing method and what can be done to resolve. Sad you didn't pick that route.

Good luck to you as well.

ExtremeFiretop commented:
@Martinski4GitHub 

I'm sorry you feel trust was lost. Wish you luck in your future endeavors.

ExtremeFiretop commented:
@Martinski4GitHub 

Submitted PR: 113 https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/113/files
It's possible the a checksum failure was the cause of my previous issues.
